lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -962,47 +962,34 @@ class MongoClientInterface {
 
     internalListObject(bucketName, params, extension, log, cb) {
         const c = this.getCollection(bucketName);
+        const stream = new MongoReadStream(c, params, params.mongifiedSearch);
+        const skip = new Skip({
+            extension,
+            gte: params.gte,
+        });
         let cbDone = false;
-        let streakLength = 0;
-        const stream = new MongoReadStream(c, params,
-                                           params.mongifiedSearch);
+
+        skip.setListingEndCb(() => {
+            stream.emit('end');
+            stream.destroy();
+        });
+        skip.setSkipRangeCb(range => {
+            // stop listing this key range
+            stream.destroy();
+            // update the new listing parameters here
+            // eslint-disable-next-line no-param-reassign
+            params.start = undefined; // 'start' is deprecated
+            // eslint-disable-next-line no-param-reassign
+            params.gt = undefined;
+            // eslint-disable-next-line no-param-reassign
+            params.gte = range;
```

Comment:
Minor: Could be a bit cleaner to create a new object and reassign these properties rather than disabling the linter.

Reply:
Done. I saw it so much in the code than I thought it is a good practice :S